### ISSUE TITLE
[Snackbar] Snackbar message uses `.button` trait but has no clear way to indicate it is a "dismiss" button

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#4685]
The Snackbar title has the UIAccessibilityTrait `.button`, but as a label its text is rarely an obvious verb. For example, in an email application the text might be, "Message saved in Drafts". It's not clear what activating that item will do, but navigating to the "Drafts" folder would not be unreasonable. I'm not sure about the best way forward with this, but it seems less than ideal.

## Reproduction steps
Turn on VoiceOver
1. Open the Catalog &rarr; Snackbar &rarr; Start Demo
1. Activate "Snackbar with Action Button"
1. Wait until VoiceOver focuses on the Snackbar label, "Snackbar Message"

## Expected behavior
UI controls marked as`.button` should have a clearly defined action, either via the surrounding context or using `accessibilityHint`. 

## Actual behavior
The Snackbar message presents the title as a button with no way for the user to know it's a "dismiss" action.

Perhaps there is an additional affordance to close/dismiss the Snackbar message for UIAccessibility?

## Platform (please complete the following information)
iPhone 5c/iOS 10.3.

**Comment by jverkoey** [comment]
Would it be reasonable for `accessibilityHint` to be set to "Dismiss" by default?
